benchmark/utils/general.hpp

```diff
@@ -297,6 +297,27 @@ std::unique_ptr<vec<ValueType>> create_vector(
     return res;
 }
 
+// Create a matrix with value indices sin(idx++)
+template <typename ValueType>
+std::unique_ptr<vec<ValueType>> create_matrix_sin(
+    std::shared_ptr<const gko::Executor> exec, gko::dim<2> size)
+{
+    struct count {
+        using itype = gko::int32;
+        itype idx;
+        count() = default;
+        itype get_next_value() { return idx++; }
+    };
+    auto my_dist = [](count &c) {
+        return std::sin(
+            static_cast<gko::remove_complex<ValueType>>(c.get_next_value()));
```

Comment:
Maybe we should consider having a specialization for complex `ValueType`s and have sinusoidal distribution for the imaginary parts as well. That may be helpful in some cases. Just an idea - we could also delay that to another PR when needed.

Reply:
The sinus benchmark is not supposed to be part of this PR (I will remove it soon, needed it for a benchmark run for a paper).
The current implementation behaves a bit different for complex values:
```
for real value types: s(i) = sin(i)
for complex value types: s(i) = complex{sin(2*i), sin(2*i + 1)}
```
Since I did not care about the complex type for my benchmark, I ignored this behaviour (since it will not make it into develop).